Uplift/AppDelegate.swift

```diff
@@ -39,57 +39,89 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
+                gyms.sort()
+                dispatchGroup.leave()
+            },
+            failure: { dispatchGroup.leave() }
+        )
+
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd"
+        // Display 4 classes from different (random) tag categories
+        NetworkManager.shared.getGymClassesForDate(date: dateFormatter.string(from: Date()),
+            completion: { classInstances in
+                // Less than 4 classes -> Use Hard coded defaults
+                if classInstances.count < 4 {
+                    dispatchGroup.leave()
+                    return
+                }
+
+                NetworkManager.shared.getTags(
+                    completion: { tags in
+                        // Sample 4 tags without replacement
+                        var randomIndeces = Set<Int>()
+                        while randomIndeces.count < 4 {
+                            let index = Int.random(in: 0..<tags.count)
+                            randomIndeces.insert(index)
+                        }
+                        // Tag categories to chose from
+                        // Each time it picks a class, remove a tag from list
+                        var tagSubset = randomIndeces.map { tags[$0] }
+                        // Iterate over unique classes that don't share the same name, Adding classes if it has a tag not yet displayed
+                        for instance in classInstances {
+                            if !classes.contains(where: { $0.className == instance.className }) {
+                                for tag in instance.tags {
+                                    if let index = tagSubset.index(of: tag) {
+                                        tagSubset.remove(at: index)
+                                        classes.append(instance)
+                                        break
+                                    }
+                                }
+                                // Chose 4 classes, can stop
+                                if classes.count >= 4 { break }
+                            }
+                        }
+
+                        classes = classes.sorted(by: { $0.className < $1.className })
+                        dispatchGroup.leave()
+                    },
+                    failure: { dispatchGroup.leave() }
+                )
+            },
+            failure: { dispatchGroup.leave() }
+        )
+
+        dispatchGroup.notify(queue: .main, execute: {
+            self.window?.rootViewController = gyms.count < 4 || classes.count < 4
+                ? OnboardingViewController()
+                : OnboardingViewController(gymNames: gyms, classes: classes)
+        })
+
+        // No Internet/Networking Failed/Networking in progress; initialize with blank VC
+        let defaultVC = UIViewController()
+        defaultVC.view.backgroundColor = .primaryWhite
+        self.window?.rootViewController = defaultVC
```

Comment:
do we have designs for if networking fails? if not, we should ask design. if we do, we should make a new task.
another question is, if the networking is slow the user will just see this blank screen right? maybe the view controller should have some loading indicator then?